### PR TITLE
Initial VLIW Infrastructure & IN/OUT Intrinsic support

### PR DESCRIPTION
**Pull request by rkabrick**
**TLDR**: 
- IN/OUT Intrinsics
- Added CTRL/DATA Signal Types
- Augmented EncodingGapPass to tell you what bits are missing/undefined
- Added CGCLI command to permit users to output raw stonecutter output from an IR file
- Fixed CSR Register Value bug 
- Added VLIW IN/OUT Metadata output 

-----------------------------------------------------------------

- Did not include code... twas quite broken but gave outline of SCVLIWPipeBuilder
- adding notes for vliw pass
- Fixed number of arguments in IN/OUT intrinsics
- Added CTRL & DATA signals (for VLIW)
- Added intrinsic error messages - 0 width args cause error printed in SCSigMap
- fixing preprocessor macro
- augmenting the EncodingGapPass to output the undefined fields
- adding changes to handle CTRL/DATA parsing
- adding command line options to cgcli to permit users to output the raw stonecutter output from an IR file.  This output DOES NOT consider any inherent linkage between ISAs and instructions.  It outputs all instructions and all ISA data.  Use for VLIW builds
- fixing line numbering to adjust to intrinsic depth
- fixing bug in erroneously setting CSR register value
- adding metadata infrastructure to IN/OUT intrinsics
- Changed IN/OUT Metadata var names for clarity
- outputs correct metadata incorrectly & fails tests :)

### CODE REVIEW

**Comment by jleidel** [comment]
Tagging Issue #348 Issue #357

**Review by jleidel — APPROVED** [review]
review complete
